Examples/BinaryBH/Main_BinaryBH.cpp

```diff
@@ -25,32 +25,44 @@ int runGRChombo(int argc, char *argv[])
     GRParmParse pp(argc - 2, argv + 2, NULL, in_file);
     SimulationParameters sim_params(pp);
 
+    BHAMR bh_amr;
+    // must be before 'setupAMRObject' to define punctures for tagging criteria
+    if (sim_params.track_punctures)
+    {
+        bh_amr.puncture_tracker.initial_setup(
+            {sim_params.bh1_params.center, sim_params.bh2_params.center},
+            sim_params.checkpoint_prefix);
+    }
+
     // The line below selects the problem that is simulated
     // (To simulate a different problem, define a new child of AMRLevel
     // and an associated LevelFactory)
-    BHAMR gr_amr;
-    DefaultLevelFactory<BinaryBHLevel> binary_bh_level_fact(gr_amr, sim_params);
-    setupAMRObject(gr_amr, binary_bh_level_fact);
+    DefaultLevelFactory<BinaryBHLevel> binary_bh_level_fact(bh_amr, sim_params);
+    setupAMRObject(bh_amr, binary_bh_level_fact);
 
     // call this after amr object setup so grids known
     // and need it to stay in scope throughout run
     AMRInterpolator<Lagrange<4>> interpolator(
-        gr_amr, sim_params.origin, sim_params.dx, sim_params.boundary_params,
+        bh_amr, sim_params.origin, sim_params.dx, sim_params.boundary_params,
         sim_params.verbosity);
-    gr_amr.set_interpolator(&interpolator);
+    bh_amr.set_interpolator(&interpolator);
+
+    // must be after interpolator is set
+    if (sim_params.track_punctures)
+        bh_amr.puncture_tracker.set_interpolator(&interpolator);
```

Comment:
After making the changes in my other comment, I think `PunctureTracker::set_interpolator` can be called by `BHAMR::set_interpolator` (I don't think it's necessary to guard this with the `track_punctures` param) and then `PunctureTracker::restart_punctures` (or similar) can be called here.